libs/pages/application/src/lib/feature/page-settings-domains-feature/crud-modal-feature/crud-modal-feature.tsx

```diff
@@ -1,126 +1,64 @@
-import { useQueryClient } from '@tanstack/react-query'
 import { type CustomDomain } from 'qovery-typescript-axios'
 import { useEffect, useMemo } from 'react'
 import { FormProvider, useForm } from 'react-hook-form'
-import { useDispatch, useSelector } from 'react-redux'
-import { useNavigate } from 'react-router-dom'
-import {
-  createCustomDomain,
-  editCustomDomain,
-  getCustomDomainsState,
-  postApplicationActionsRedeploy,
-} from '@qovery/domains/application'
-import { useLinks } from '@qovery/domains/services/feature'
-import { getServiceType } from '@qovery/shared/enums'
-import { type ApplicationEntity, type LoadingStatus } from '@qovery/shared/interfaces'
-import { DEPLOYMENT_LOGS_URL, ENVIRONMENT_LOGS_URL } from '@qovery/shared/routes'
+import { type AnyService } from '@qovery/domains/services/data-access'
+import { useCreateCustomDomain, useEditCustomDomain, useLinks } from '@qovery/domains/services/feature'
 import { useModal } from '@qovery/shared/ui'
-import { type AppDispatch, type RootState } from '@qovery/state/store'
-import { queries } from '@qovery/state/util-queries'
 import CrudModal from '../../../ui/page-settings-domains/crud-modal/crud-modal'
 
 export interface CrudModalFeatureProps {
-  organizationId: string
-  projectId: string
   customDomain?: CustomDomain
-  application?: ApplicationEntity
+  service?: AnyService
```

Comment:
It should be better to limit `serviceType` here so you are guarantee about the `serviceType` everywhere and you can handle wrong serviceType in `PageSettingsDomainsFeature` using `redirect` or whatever.
You can also handle `undefined` service upfront